### PR TITLE
Issue #3177 Replace AssertJ's deprecated asList() DSL method with asInstanceOf in three test files

### PR DESCRIPTION
**Pull request by Flowers2Algernon**
… WildflyJARHealthCheckEnricherTest.java,WildflyJARGeneratorTest.java and VertxHealthCheckEnricherTest.java

## Description

Fixes # 3177, replace deprecated `asList()` method to `asInstanceOf(InstanceOfAssertFactories.list(type.class))` method

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3186 (_2024-06-21T12:23:24Z_)

:gear: [JKube E2E Tests (9613536689)](https://github.com/eclipse-jkube/ci/actions/runs/9613536689) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3186)](https://github.com/eclipse-jkube/ci/actions/runs/9613536689/job/26516476183)

File: jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/enricher/VertxHealthCheckEnricherTest.java [hunk 1]
```diff
@@ -540,9 +541,10 @@ void withExecTypeReadinessUsingConfig_shouldDisableLiveness() throws Exception {
 
         Probe readinessProbe = enricher.getReadinessProbe();
         assertThat(readinessProbe).isNotNull()
-            .extracting(Probe::getExec)
-            .extracting(ExecAction::getCommand).asList()
-            .hasSize(3);
+                .extracting(Probe::getExec)
+                .extracting(ExecAction::getCommand)
+                .asInstanceOf(InstanceOfAssertFactories.list(String.class))
+                .hasSize(3);
```
Comment (rohanKanojia) [inline jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/enricher/VertxHealthCheckEnricherTest.java:547]: could you please revert formatting changes? Have you configured your IDE to use [eclipse formatter settings](https://eclipse.dev/jkube/contributing/#intellij-idea-setup)?

Comment (Flowers2Algernon) [reply jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/enricher/VertxHealthCheckEnricherTest.java:547]: OK, I will setting my eclipse to make these formatting changes not happen,  but after my settings done, should I make a new pr or continue with this pr?

Comment (rohanKanojia) [reply jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/enricher/VertxHealthCheckEnricherTest.java:547]: No need to create a new PR. You can make changes and follow these steps: 
```
# After making your changes,add them to git index
git add file/you/changed
# Amend previous commit
git commit --amend
# Force push to your existing branch
git push origin master -f
```

Comment (rohanKanojia) [reply jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/enricher/VertxHealthCheckEnricherTest.java]: @Flowers2Algernon : I see you're committed changes in master branch. It's advised to create a new branch for a new change request.

Comment (Flowers2Algernon) [reply jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/enricher/VertxHealthCheckEnricherTest.java]: thank you for your kind advice, just now I reset my git version and now I will follow your guide to do this, thx again

Comment (rohanKanojia) [reply jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/enricher/VertxHealthCheckEnricherTest.java]: @Flowers2Algernon : No problem!

**Pull request closed without merging**